client/src/app/core/repositories/agenda/list-of-speakers-repository.service.ts

```diff
@@ -228,6 +239,96 @@ export class ListOfSpeakersRepositoryService extends BaseHasContentObjectReposit
         return !this.getViewModelList().some(list => list.hasSpeakerSpoken(speaker));
     }
 
+    /**
+     * List every speaker only once, who has spoken
+     *
+     * @returns A list with all different speakers.
+     */
+    public getAllFirstContributions(): ViewSpeaker[] {
+        const speakers: ViewSpeaker[] = this.getViewModelList().flatMap(
+            (los: ViewListOfSpeakers) => los.finishedSpeakers
+        );
+        const firstContributions: ViewSpeaker[] = [];
+        for (const speaker of speakers) {
+            if (!firstContributions.find(s => s.user_id === speaker.user_id)) {
+                firstContributions.push(speaker);
+            }
+        }
+        return firstContributions;
+    }
+
+    /**
+     * Maps structure-level to speaker.
+     *
+     * @returns A map, which keys are structure-levels and values a `SpeakingTimeStructureLevelObject`.
+     */
+    public getSpeakingTimeStructureLevelRelation(): Map<string, SpeakingTimeStructureLevelObject> {
```

Comment:
This function is just used once and only the values are needed. So you can return just a list of `SpeakingTimeStructureLevelObject`